### PR TITLE
Fix bug that disabled game-based minetest.conf default setting

### PR DESCRIPTION
**Pull request by est31**
commit f6e4c5d9cf459e8278a76a2beaee59732e841458 caused this.

### CODE REVIEW

**Comment by ShadowNinja** [comment]
This code also has to run when initializeWorld is called by the main menu so that mapgen parameter defaults are initialized.

File: src/subgame.cpp [hunk 1]
```diff
@@ -267,23 +267,22 @@ std::vector<WorldSpec> getAvailableWorlds()
...
 }
 
-bool initializeWorld(const std::string &path, const std::string &gameid)
+bool loadGameConfAndInitWorld(const std::string &path, const SubgameSpec &gamespec)
 {
+	// Override defaults with those provided by the game.
+	Settings game_defaults;
+	getGameMinetestConfig(gamespec.path, game_defaults);
+	override_default_settings(g_settings, &game_defaults);
+
```
Comment (ShadowNinja) [inline src/subgame.cpp]: This should stay where it was below.

Comment (est31) [inline src/subgame.cpp]: its not part of the world initialisation.

**Comment by nerzhul** [comment]
jenkins build please

**Comment by ShadowNinja** [comment]
538036d004f914f31c276090d7735d61139fc58e

**Pull request closed without merging**